src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -0,0 +1,353 @@
+import SearchPage from '../../../pages/SearchPage';
+import ProfilePage from '../../../pages/ProfilePage';
+import IOURequestPage from '../../../pages/iou/IOURequestPage';
+import IOUBillPage from '../../../pages/iou/IOUBillPage';
+
+const Drawer = createDrawerNavigator();
+const RootStack = createStackNavigator();
+
+const defaultSubRouteOptions = {
+    cardStyle: styles.navigationScreenCardStyle,
+    headerShown: false,
+};
+
+const propTypes = {
+    network: PropTypes.shape({isOffline: PropTypes.bool}),
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    network: {isOffline: true},
+};
+
+class AuthScreens extends React.Component {
+    constructor(props) {
+        super(props);
+
+        Timing.start(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+        Timing.start(CONST.TIMING.HOMEPAGE_REPORTS_LOADED);
+    }
+
+    componentDidMount() {
+        NetworkConnection.listenForReconnect();
+        PusherConnectionManager.init();
+        Pusher.init({
+            appKey: CONFIG.PUSHER.APP_KEY,
+            cluster: CONFIG.PUSHER.CLUSTER,
+            authEndpoint: `${CONFIG.EXPENSIFY.URL_API_ROOT}api?command=Push_Authenticate`,
+        }).then(subscribeToReportCommentEvents);
+
+        // Fetch some data we need on initialization
+        NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.PRIORITY_MODE, 'default');
+        PersonalDetails.fetch();
+        PersonalDetails.fetchTimezone();
+        getBetas();
+        fetchAllReports(true, true);
+        fetchCountryCodeByRequestIP();
+        UnreadIndicatorUpdater.listenForReportChanges();
+
+        // Refresh the personal details, timezone and betas every 30 minutes
+        // There is no pusher event that sends updated personal details data yet
+        // See https://github.com/Expensify/ReactNativeChat/issues/468
+        this.interval = setInterval(() => {
+            if (this.props.network.isOffline) {
+                return;
+            }
+            PersonalDetails.fetch();
+            PersonalDetails.fetchTimezone();
+            getBetas();
+        }, 1000 * 60 * 30);
+
+        Timing.end(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+
+        // Listen for the Command+K key being pressed so the focus can be given to the chat switcher
+        KeyboardShortcut.subscribe('K', () => {
+            redirect(ROUTES.SEARCH);
+        }, ['meta'], true);
+    }
+
+    componentWillUnmount() {
+        KeyboardShortcut.unsubscribe('K');
+        NetworkConnection.stopListeningForReconnect();
+        clearInterval(this.interval);
+        this.interval = null;
+    }
+
+    render() {
+        const modalScreenOptions = {
+            headerShown: false,
+            cardStyle: getNavigationModalCardStyle(this.props.isSmallScreenWidth),
+            gestureDirection: 'horizontal',
+        };
+
+        if (this.props.isSmallScreenWidth) {
+            modalScreenOptions.cardStyleInterpolator = CardStyleInterpolators.forScaleFromCenterAndroid;
```

Comment:
you might want to give certain stacks specific animations instead of giving all of them this one. here's an example:

https://user-images.githubusercontent.com/12075600/110578828-73640680-811a-11eb-8d6b-5f0234d65bf7.mov

Reply:
Yeah, I think this is probably something we will hash out later. I do know whatever we want to do we have a preference to make things consistent across the platforms.